pysimplesoap/client.py

```diff
@@ -142,17 +148,64 @@ def __init__(self, location=None, action=None, namespace=None,
             f.close()
         self.cacert = cacert
 
-        # Create HTTP wrapper
-        Http = get_Http()
-        self.http = Http(timeout=timeout, cacert=cacert, proxy=proxy, sessions=sessions)
+        # if username and password add basic authentication
         if username and password:
-            if hasattr(self.http, 'add_credentials'):
-                self.http.add_credentials(username, password)
-        if cert and key_file:
-            if hasattr(self.http, 'add_certificate'):
-                self.http.add_certificate(key=key_file, cert=cert, domain='')
+            auth_string = "{0}:{1}".format(username, password)
+            auth_header = make_headers(basic_auth=auth_string)
+            self.http_headers.update(auth_header)
 
 
+        # Create HTTP wrapper with connection pooling if num_pools specified
```

Comment:
As far as I can tell, get_http should return the transport class object. Please refactor this such that http = get_Http( ) returns a url3 transport when requested. You may need to pass in an indicator that this is a url_transport3.